react-app/src/api/govAPI.ts

```diff
@@ -14,20 +16,100 @@ import {
 } from "../utils/coin";
 import { SignedTx, useCosmosAPI } from "./cosmosAPI";
 
+interface DepositParams {
+  minDeposit: BigNumberCoin;
+  /**
+   * Maximum deposit period in days
+   */
+  maxDepositPeriod: number;
+}
+
+/**
+ * TallyParams defines the params for tallying votes on governance proposals.
+ * results are returned in percentages
+ */
+export interface TallyParams {
+  /**
+   * Minimum percentage of total stake needed to vote for a result to be
+   *  considered valid.
+   */
+  quorum: number;
+  /** Minimum proportion of Yes votes for proposal to pass. Default value: 0.5. */
+  threshold: number;
+  /**
+   * Minimum value of Veto votes to Total votes ratio for proposal to be
+   *  vetoed. Default value: 1/3.
+   */
+  vetoThreshold: number;
+}
+
+interface VotingParams {
+  /**
+   * Maximum voting period in days
+   */
+  votingPeriod: number;
+}
+
+export interface GovParams {
+  deposit: DepositParams;
+  tally: TallyParams;
+  voting: VotingParams;
+}
+
 interface IGovAPI {
-  getMinDepositParams(): Promise<BigNumberCoin>;
   signSubmitProposalTx(
     proposal: ProposalContentBody,
     initialDeposit: BigNumber,
     memo?: string
   ): Promise<SignedTx>;
+  getMinDepositParams(): Promise<BigNumberCoin>;
+  getAllParams(): Promise<GovParams>;
 }
 
+const CoinMinimalDenom = Config.chainInfo.currency.coinMinimalDenom;
+const CoinDenom = Config.chainInfo.currency.coinDenom;
+
+/**
+ * Convert cosmjs Duration to days
+ * return 0 if duration is undefined
+ */
+const convertDurationToDays = (duration: Duration | undefined) => {
+  if (!duration) {
+    return 0;
+  }
+  return duration.seconds.div(60 * 60 * 24).toInt();
+};
```

Comment:
I think this should be handled in the UI component, also can we use datefns for this?

Reply:
date-fns dont have secondsToDay :sosad